frame/staking/src/pallet/impls.rs

```diff
@@ -841,13 +864,21 @@ impl<T: Config> Pallet<T> {
 	/// Get the targets for an upcoming npos election.
 	///
 	/// This function is self-weighing as [`DispatchClass::Mandatory`].
-	pub fn get_npos_targets(maybe_max_len: Option<usize>) -> Vec<T::AccountId> {
-		let max_allowed_len = maybe_max_len.unwrap_or_else(|| T::TargetList::count() as usize);
-		let mut all_targets = Vec::<T::AccountId>::with_capacity(max_allowed_len);
+	pub fn get_npos_targets(target_bounds: DataProviderBounds) -> Vec<T::AccountId> {
+		let max_allowed_len = {
+			let all_target_count = T::TargetList::count();
+			target_bounds
+				.count
+				.unwrap_or(all_target_count.into())
```

Comment:
I think this chain of calls is a bit hard to follow, but not sure how to improve it either.